core/templates/dev/head/domain/exploration/ExplorationObjectFactory.js

```diff
@@ -23,6 +23,8 @@ oppia.factory('ExplorationObjectFactory', [
   function(
       INTERACTION_SPECS, INTERACTION_DISPLAY_MODE_INLINE, StateObjectFactory,
       UrlInterpolationService) {
+    var currentExploration = null;
```

Comment:
(Generally speaking, it's good to move the logic to services and keep the controllers relatively thin, since services are easier to test. You don't necessarily need to do this fully now, but there needs to be some way to keep the object factory stateless.)